### ISSUE TITLE
Usage of struct sockaddr

### ISSUE THREAD

**Issue by jpboivin** [ngtcp2/ngtcp2#276]
Hello, 

So far, ngtcp2 has been extremely easy to port to the ~10 platforms supported where I work because it doesn't rely on anything outside of plain C. Platform-specific details (like sockets, cryptography, file system) are normally kept out of the library. 

Since the following commit https://github.com/ngtcp2/ngtcp2/commit/f18bb8544137d9d796902e08b0942d814f8a7474, the path holds an `struct sockaddr` instead of the previous opaque structure. I can understand the better comparisons (I had to deal with padding issues with the custom structure we introduced), but I'm wondering if ngtcp2 shouldn't instead introduce its own structure for that? Although `struct sockaddr` is normally supported (and its IPv4/IPv6 equivalents), not all platforms have them.

**Comment by tatsuhiro-t** [comment]
It might be nice to define struct sockaddr family for those platforms which lack these structs.
For platforms that have struct sockaddr, using them is extremely easy because they are provided by syscall and no conversion is necessary.

**Comment by jpboivin** [comment]
I wouldn't mind defining them, but I wouldn't like defining them to generic `struct sockaddr` either. That's why maybe a ngtcp2_sockaddr would make sense (maybe a type alias on the platforms that support it). Currently, I would need to define these types (and the `AF_INET` defines) in the ngtcp2.h header, which I find hacky.

**Comment by tatsuhiro-t** [comment]
ngtcp2_sockaddr (and sockaddr_* variants) might be a good idea.
I think enumerating platforms which have struct sockaddr (or platform not having them) is also tedious (we cannot use config.h here), it might be easier to add macro to indicate that it uses generic version sockaddr, like so:

```c
#ifdef USE_GENERIC_SOCKADDR
struct ngtcp2_sockaddr {
  ...
} ngtcp2_sockaddr;
...
#else /* !USE_GENERIC_SOCKADDR */
typedef struct sockaddr ngtcp2_sockaddr;
typedef struct sockaddr_storage ngtcp2_sockaddr_storage;
#endif
```

**Comment by tatsuhiro-t** [comment]
Fixed by #354
